test/unit/cfg_errors_test.lua

```diff
@@ -263,6 +278,71 @@ g.test_console_sock = function()
     end
 end
 
+-- resolve_dns ----------------------------------------------------------------
+-------------------------------------------------------------------------------
+
+g.test_dns_not_resolved = function()
+    g.mock_membership()
+    dns_resolved = false
+
+    check_error('Can not ping myself',
+        cartridge.cfg, {
+            workdir = g.tempdir,
+            advertise_uri = 'localhost:13000',
+            roles = {},
+        }
+    )
+    dns_resolved = true
+end
+
+g.test_dns_resolve_after_wait = function()
+    g.mock_membership()
+    dns_resolved = false
+
+    fiber.create(function()
+        -- sleep time < default probe_uri_timeout
+        fiber.sleep(0.3)
+        dns_resolved = true
+    end)
+
+    -- default probe_uri_timeout: 1.0s
+    local ok, err = cartridge.cfg({
+        workdir = g.tempdir,
+        advertise_uri = 'localhost:13001',
+        roles = {},
+    })
```

Comment:
You were going to make default 0. I suggest passing timeout explicitly with `os.setenv`